### ISSUE TITLE
Integrity hash generated on a different file than the one built

### ISSUE THREAD

**Issue by Samy-Belarbi** [lhapaipai/vite-bundle#65]
Hey, it's me again ! 😬

Thanks again for your work.

I implemented, in my company app, your new [SRI Feature](https://github.com/lhapaipai/vite-bundle/pull/52) and i am facing an error :

```Failed to find a valid digest in the 'integrity' attribute for resource 'example.js' with computed SHA-384 integrity 'FFnZ[...]'. The resource has been blocked.```

So i tried to locally generate a hash for my `example.js`, and it does not match the one generated in the `entrypoints.json` (but it matches the integrity hash given by my browser, shared in the error message).

After some few investigations, it looks like the chunk `example.js` given to the `generateHash()` in `vite-plugin-symfony` isn't exactly the same as the `example.js` built in my build folder. The one given has some `__VITE_PRELOAD__` inside his code near where i do some lazy imports with React.

If you need some more informations please let me know.

Have a great day !

**Comment by lhapaipai** [comment]
hi @Samy-Belarbi, the Hash is generated inside the rollupHook : `generateBundle` (getFileInfos function https://github.com/lhapaipai/vite-plugin-symfony/blob/main/src/index.ts#L217)
Have you try to order vite-plugin-symfony at the end of your plugins list, after the plugin who transform `__VITE_PRELOAD__` ? https://vitejs.dev/guide/using-plugins.html#enforcing-plugin-ordering.
else can you give me a reproductive example ?
good afternoon !

**Comment by PaulSenon** [comment]
Have the same issue, and symfonyPlugin() is my only plugin. So I can't reorder anything. 

Here is the vite config:
```
import { defineConfig } from 'vite';
import symfonyPlugin from 'vite-plugin-symfony';

export default defineConfig(() => {
  return {
    plugins: [
      symfonyPlugin({
        sriAlgorithm: "sha256",
      }),
    ],
    resolve: {
      alias: [
        {
          find: '@js',
          replacement: fileURLToPath(
            new URL('./path/to/my/js/asset/root', import.meta.url),
          ),
        },
      ],
    },
    build: {
      rollupOptions: {
        input: {
          // JS
          entrypoint1_js: "my/path/to/my/js/entrypoint1",
          entrypoint2_js: "my/path/to/my/js/entrypoint2",

          // CSS
          entrypoint1_css: "my/path/to/my/sass/entrypoint1",
          entrypoint2_css: "my/path/to/my/sass/entrypoint2",
        },
      },
    },
  };
});
```

From my investigation, yes it is because the hash is computed before `__VITE_PRELOAD__` are replaced. But I'm new to this and don't know what is supposed to handle this.

Hope it helps a bit.

I'll update this thread if I ever spend more time on it :)

**Comment by darrylhein** [comment]
I'm having the same issue.

**Comment by lhapaipai** [comment]
fixed with `vite-plugin-symfony` v4.2.0 : https://github.com/lhapaipai/vite-plugin-symfony/commit/ce2783b1024d28d9a931a85e4de754fa84cd553b

documentation on : https://symfony-vite.pentatrion.com/config/vite-plugin-symfony.html#enforcepluginorderingposition

in your case you will have to leave the `enforcePluginOrderingPosition` to `true` (which is the default value).

**Comment by glxgxr** [comment]
I am experiencing this issue today, but only with the javascript file. This is my setup:

composer.json
```json
     "pentatrion/vite-bundle": "^v6.3.0",
```

package.json
```json
    "vite": "^5.0.10",
    "vite-plugin-static-copy": "^1.0.0",
    "vite-plugin-symfony": "^6.3.0",
```

vite.config.mjs
```js
import {defineConfig, loadEnv} from 'vite';
import symfonyPlugin from 'vite-plugin-symfony';
import vue from '@vitejs/plugin-vue';
import nodeResolve from '@rollup/plugin-node-resolve';
import {viteStaticCopy} from 'vite-plugin-static-copy';

//eslint-disable-next-line no-unused-vars
export default defineConfig(({command, mode}) => {
    //eslint-disable-next-line no-undef
    const env = loadEnv(mode, process.cwd(), '');

    return {
        base: '/build/',
        plugins: [
            nodeResolve(),
            vue(),
            symfonyPlugin({
                servePublic: 'public',
                sriAlgorithm: "sha256"
            }),
            viteStaticCopy({
                targets: [
                    {
                        src: './assets/images',
                        dest: './',
                    },
                    {
                        src: './assets/videos',
                        dest: './',
                    },
                ],
            }),
        ],
        build: {
            root: './assets',
            rollupOptions: {
                input: {
                    appScripts: './assets/js/app.js',
                    leaflet: './assets/js/leaflet.js',
                },
                output: {
                    manualChunks: {
                        vue: ['vue']
                    }
                }
            },
            define: {
                __APP_ENV__: env.APP_ENV || 'dev',
            },
            outDir: 'public/build',
        },
        server: {
          // ...
        },
    };
});

```

All hashes for the css files seem to work fine, except the one for javascript:

> Failed to find a valid digest in the 'integrity' attribute for resource 'https://app.blabla.local/build/assets/appScripts-EqTxXgdP.js' with computed SHA-256 integrity 'ljVss2YLNlzXj3vEVZ....'. The resource has been blocked.

The hash just doesn't match with the one in the `manifest.json`. Any ideas?
